lib/Jsrt/JsrtDebugManager.cpp

```diff
@@ -154,6 +154,18 @@ HRESULT JsrtDebugManager::DbgRegisterFunction(Js::ScriptContext* scriptContext,
         {
             utf8SourceInfo->SetDebugDocument(debugDocument);
         }
+
+        // Only raise an event in the case where we have a registered callback and we are not
+        // currently in the middle of a source reparse (e.g. attaching/detaching).
```

Comment:
>middle of a source reparse (e.g. attaching/detaching). [](start = 28, length = 54)

Please explain why we are adding isParsingSource check, will help somebody understand the code/logic when we are not around.